lib/semantics/check-do.cc

```diff
@@ -319,14 +319,42 @@ class DoConcurrentLabelEnforce {
   parser::CharBlock doConcurrentSourcePosition_{nullptr};
 };  // class DoConcurrentLabelEnforce
 
-using CS = std::vector<const Symbol *>;
+// Class for enforcing C1130
+class DoConcurrentVariableEnforce {
+public:
+  DoConcurrentVariableEnforce(
+      SemanticsContext &context, parser::CharBlock doConcurrentSourcePosition)
+    : context_{context}, doConcurrentSourcePosition_{
+                             doConcurrentSourcePosition} {}
 
-struct GatherSymbols {
-  CS symbols;
-  template<typename T> constexpr bool Pre(const T &) { return true; }
-  template<typename T> constexpr void Post(const T &) {}
-  void Post(const parser::Name &name) { symbols.push_back(name.symbol); }
-};
+  template<typename T> bool Pre(const T &) { return true; }
+  template<typename T> void Post(const T &) {}
+
+  // Check to see if the name is a variable from an enclosing scope
+  void Post(const parser::Name &name) {
+    if (const Symbol * symbol{name.symbol}) {
+      if (IsVariableName(*symbol)) {
+        const Scope &variableScope{symbol->owner()};
+        if (DoesScopeContain(&variableScope, GetBlockScope())) {
+          context_.Say(name.source,
+              "Variable '%s' from enclosing scope in a DEFAULT(NONE) DO "
+              "CONCURRENT, must appear in a locality-spec"_err_en_US,
+              name.source);
+        }
+      }
+    }
+  }
+
+private:
+  const Scope &GetBlockScope() {
```

Comment:
This function is going to be called once for every variable name in the loop and it's going to return the same thing every time.

Reply:
Thanks, Peter.  I'll fix that.

Reply:
@klausler , can you please look at my latest commit and see if it satisfies your concerns?  Note that this comment is blocking my ability to merge.